csrc/driver_api.cpp

```diff
@@ -5,22 +5,33 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 // clang-format on
-#include <driver_api.h>
-#include <sys_utils.h>
-#include <utils.h>
+#include <iostream>
 
 #include <cuda.h>
+#include <cuda_runtime.h>
 
-#include <iostream>
-
+#include <cuda_utils.h>
+#include <driver_api.h>
 #include <exceptions.h>
+#include <sys_utils.h>
+#include <utils.h>
+
+namespace nvfuser {
+PFN_cuStreamWriteValue32 cuStreamWriteValue32;
+} // namespace nvfuser
 
 namespace {
 
 class CUDADriverAPIDynamicLoader : public nvfuser::LibraryLoader {
  public:
   CUDADriverAPIDynamicLoader() {
     setFilename("libcuda.so");
+
+    // https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#using-the-runtime-api
+    NVFUSER_CUDA_RT_SAFE_CALL(cudaGetDriverEntryPoint(
+        "cuStreamWriteValue32",
+        reinterpret_cast<void**>(&nvfuser::cuStreamWriteValue32),
+        cudaEnableDefault));
   }
 } loader;
```

Comment:
I'm quite worried about the order of global initializers. Any suggestions, @zasdfgbnm ?